client/my-sites/people/people-section-nav/index.jsx

```diff
@@ -53,11 +53,13 @@ let PeopleNavTabs = React.createClass( {
 	}
 } );
 
-module.exports = React.createClass( {
+class PeopleSectionNav extends Component {
 
-	displayName: 'PeopleSectionNav',
+	canSearch() {
```

Comment:
The `site.versionCompare( '3.7.0-beta', '<' )` regression for Jetpack sites exists in this component, too.

Side note: if you are to connect this component, perhaps it could also be a good idea to change some of the props (`site`, `siteSlug`, `isJetpackSite`) to be connected props instead of being passed from the parent component.